src/System.Threading/tests/ThreadLocalTests.cs

```diff
@@ -0,0 +1,418 @@
+            }
+            catch (ArgumentNullException)
+            {
+                // No other exception should be thrown.
+                // With a previous issue, if the constructor threw an exception, the finalizer would throw an exception as well.
+            }
+        }
+
+        /// <summary>Tests for the ToString.</summary>
+        /// <returns>True if the tests succeeds, false otherwise.</returns>
+        [Fact]
+        public static void RunThreadLocalTest2_ToString()
+        {
+            ThreadLocal<object> tlocal = new ThreadLocal<object>(() => (object)1);
+            if (tlocal.ToString() != 1.ToString())
+            {
+                Assert.True(false,
+                    string.Format("RunThreadLocalTest2_ToString: > test failed - Unexpected return value from ToString(); Actual={0}, Expected={1}.",
+                        tlocal.ToString(), 1.ToString()));
+            }
+        }
+
+        /// <summary>Tests for the Initialized property.</summary>
+        /// <returns>True if the tests succeeds, false otherwise.</returns>
+        [Fact]
+        public static void RunThreadLocalTest3_IsValueCreated()
+        {
+            ThreadLocal<string> tlocal = new ThreadLocal<string>(() => "Test");
+            if (tlocal.IsValueCreated)
+            {
+                Assert.True(false, "RunThreadLocalTest3_IsValueCreated: > test failed - expected ThreadLocal to be uninitialized.");
+            }
+            string temp = tlocal.Value;
+            if (!tlocal.IsValueCreated)
+            {
+                Assert.True(false, "RunThreadLocalTest3_IsValueCreated: > test failed - expected ThreadLocal to be initialized.");
+            }
+        }
+
+        [Fact]
+        public static void RunThreadLocalTest4_Value()
+        {
+            ThreadLocal<string> tlocal = null;
+
+            // different threads call Value
+            int numOfThreads = 10;
+            Task[] threads = new Task[numOfThreads];
+            object alock = new object();
+            List<string> seenValuesFromAllThreads = new List<string>();
+            int counter = 0;
+            tlocal = new ThreadLocal<string>(() => (++counter).ToString());
+            //CancellationToken ct = new CancellationToken();
+            for (int i = 0; i < threads.Length; ++i)
+            {
+                // We are creating the task using TaskCreationOptions.LongRunning because...
+                // there is no guarantee that the Task will be created on another thread.
+                // There is also no guarantee that using this TaskCreationOption will force
+                // it to be run on another thread.
+                threads[i] = new Task(() =>
+                {
+                    string value = tlocal.Value;
+                    Debug.WriteLine("Val: " + value);
+                    seenValuesFromAllThreads.Add(value);
+                }, TaskCreationOptions.LongRunning);
+                threads[i].Start();
```

Comment:
Either this should use Task.Factory.StartNew(..., CancellationToken.None, TaskCreationOptions.LongRunning, TaskScheduler.Default), or TaskScheduler.Default should be passed to Start(TaskScheduler.Default).